### PR TITLE
Push all native sol operations into the program layer

### PR DESCRIPTION
**Pull request by ChewingGlass**

### CODE REVIEW

File: packages/react/src/contexts/accountContext.tsx [hunk 1]
```diff
@@ -30,11 +29,9 @@ export const AccountProvider: FC<IAccountProviderProps> = ({
     });
   }, [connection]);
 
-  if (cache) {
-    return (
-      <AccountContext.Provider value={cache}>{children}</AccountContext.Provider>
-    );
-  }
+  return (
```
Comment (ChewingGlass) [inline packages/react/src/contexts/accountContext.tsx:32]: This was required or Server Side Rendering always returns nothing

File: packages/spl-token-bonding/src/index.ts [hunk 2]
```diff
@@ -1471,29 +1435,46 @@ export class SplTokenBonding extends AnchorSdk<SplTokenBondingIDL> {
...
-    instructions.push(await this.instruction.buyV0(args, accounts));
-    instructions.push(...lastInstructions);
+
+    if (isNative) {
+      instructions.push(await this.instruction.buyNativeV0(args, { 
+        accounts: {
+          common,
+          state: state.publicKey,
+          wrappedSolMint: state.wrappedSolMint,
+          mintAuthority: (await SplTokenBonding.wrappedSolMintAuthorityKey(this.programId))[0],
```
Comment (bryzettler) [inline packages/spl-token-bonding/src/index.ts:1458]: Would it make sense to just have 
```javascript
SplTokenBonding.wrappedSolMintAuthorityKey
```
return the item at the ```[0]``` index? is there ever a time we dont do that?

**Review by bryzettler — APPROVED** [review]
Everything looks good to me, would make sure to think about any edge cases and have tests.
